bin/check-ssl-hsts-preloadable.rb

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env ruby
+# encoding: UTF-8
+#  check-ssl-hsts-preloadable.rb
+#
+# DESCRIPTION:
+#   Checks a domain against the chromium HSTS API returning errors/warnings if the domain is preloadable
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#
+# USAGE:
+#   # Basic usage
+#   check-ssl-hsts-preloadable.rb -d <domain_name>
+#
+# LICENSE:
+#   Copyright 2017 Rowan Wookey <admin@rwky.net>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE for
+#   details.
+#   
+#   Inspired by https://github.com/sensu-plugins/sensu-plugins-ssl/blob/master/bin/check-ssl-qualys.rb Copyright 2015 William Cooke <will@bruisyard.eu>
+#
+
+require 'sensu-plugin/check/cli'
+require 'json'
+require 'net/http'
+
+class CheckSSLHSTSPreload < Sensu::Plugin::Check::CLI
+
+  API_URL = 'https://hstspreload.org/api/v2/preloadable'.freeze
+
+  option :domain,
+         description: 'The domain to run the test against',
+         short: '-d DOMAIN',
+         long: '--domain DOMAIN',
+         required: true
+
+  def run
+    uri       = URI(API_URL)
+    uri.query = URI.encode_www_form({ domain: config[:domain] })
+    response  = Net::HTTP.get_response(uri)
+    warning 'Bad response recieved from API' unless response.is_a?(Net::HTTPSuccess)
```

Comment:
I think we might want to handle a redirection rather than simply saying its bad.